receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go

```diff
@@ -42,8 +43,13 @@ func getContainerMetrics(stats ContainerStats) ECSMetrics {
 	m.SystemCPUUsage = *stats.CPU.SystemCPUUsage
 	m.CPUUtilized = cpuUtilized
 
-	m.NetworkRateRxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
-	m.NetworkRateTxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
+	if stats.NetworkRate == (NetworkRateStats{}) {
+		m.NetworkRateRxBytesPerSecond = floatZero
+		m.NetworkRateTxBytesPerSecond = floatZero
+	} else {
+		m.NetworkRateRxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
+		m.NetworkRateTxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
+	}
 
```

Comment:
Can I learn where did the NPE throw? `*stats.NetworkRate.TxBytesPerSecond` shouldn't give NPE when `*stats.NetworkRate` is empty structure.

Reply:
It's in the same line you mentioned. To me, when we have an empty structure, we don't have references for `RxBytesPerSecond` and `TxBytesPerSecond`. When we are dereferencing values from these pointers, it's throwing a null pointer exception. At least, thats my understanding.